### PR TITLE
Migrate the rest of the alternate configurations builds from Buildkite

### PR DESCRIPTION
**Pull request by GMNGeoffrey**
I'll disable and delete the Buildkite pipeline after this goes in.

Part of https://github.com/iree-org/iree/issues/9855

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -581,6 +581,91 @@ jobs:
             gcr.io/iree-oss/base@sha256:605d86ccf4197e978a24867fabb7fc100334c926b067ee0518e46d0a4396e206 \
             ./build_tools/cmake/build_and_test_tsan.sh
 
+  small_runtime:
+    needs: setup
+    if: needs.setup.outputs.should-run == 'true'
+    runs-on: ubuntu-20.04-64core
```
Comment (ScottTodd) [inline .github/workflows/ci.yml:587]: I was going to suggest grouping some of these jobs if the overhead of checking out the repo and its submodules exceeded the time to run the build, but it looks like those steps are fast enough on these machines (2s and 11s, then a 1 minute build)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -581,6 +581,91 @@ jobs:
...
+            "${BUILD_DIR}"
+      - name: "Testing runtime"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            --env IREE_VULKAN_DISABLE=1 \
+            gcr.io/iree-oss/base@sha256:605d86ccf4197e978a24867fabb7fc100334c926b067ee0518e46d0a4396e206 \
+            ./build_tools/cmake/ctest_all.sh \
+            "${BUILD_DIR}"
+
+  gcc:
```
Comment (ScottTodd) [inline .github/workflows/ci.yml:611]: Add these new jobs to `summary` at the bottom of the file?

**Comment by GMNGeoffrey** [comment]
We'll see how much ccache speeds up the gcc build, but I think I'll start it as postsubmit-only

File: .github/workflows/ci.yml (hunk 2 above)
Comment (GMNGeoffrey) [reply .github/workflows/ci.yml:611]: Whoops! Thanks for remembering

**Comment by GMNGeoffrey** [comment]
Hmmm looks like gcc is failing to build tint. The build script it was using before didn't turn on Web GPU. I'll have to investigate why clang doesn't hit the same unused diagnostics

**Comment by GMNGeoffrey** [comment]
Ah actually the question is what's setting Werror for gcc. We explicitly don't set that: https://github.com/iree-org/iree/blob/0246972fa405f2b777d9a6dc7589ce143f380fc4/build_tools/cmake/iree_copts.cmake#L204

**Comment by ScottTodd** [comment]
Interesting. I don't think we care too much about Tint + gcc, unless we need that somehow for python packaging? That particular error could be worth filing upstream and fixing. For now just figuring out the compiler flags to ignore the problem might be fine.

**Comment by GMNGeoffrey** [comment]
It looks like tint is the one setting Werror: https://dawn.googlesource.com/tint/+/refs/heads/main/CMakeLists.txt#330

**Comment by GMNGeoffrey** [comment]
Filed https://github.com/iree-org/iree/issues/11394 and disabled Web GPU for GCC

File: .github/workflows/ci.yml (hunk 1 above)
Comment (GMNGeoffrey) [reply .github/workflows/ci.yml:587]: Yeah checkouts are way faster on GitHub runners. We should be able to optimize them on our runners too somehow. I'm actually surprised at how slow they are given that we're fetching with depth=1. I think I tried using our own submodule checkout with `--jobs=8`, but it didn't seem to help. Reference repos or local mirrors might help

**Review by pzread — APPROVED** [review]

**Comment by GMNGeoffrey** [comment]
I think gcc is fast enough to go on presubmit, especially given all the other things that are slower right now (:cry:) and that it should speed up significantly with ccache (:slightly_smiling_face:)

**Pull request merged**